File: src/main/java/com/googongill/aditory/service/dto/link/LinkListResult.java
package com.googongill.aditory.service.dto.link;

import lombok.Builder;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Builder
public class LinkListResult {
    @Builder.Default
    private List<LinkInfo> linkList = new ArrayList<>();

    public static LinkListResult of(List<LinkInfo> linkInfoList) {
        return LinkListResult.builder()
                .linkList(linkInfoList)
                .build();
    }
}

File: src/main/java/com/googongill/aditory/service/dto/link/LinkListResult.kt
package com.googongill.aditory.service.dto.link

data class LinkListResult(
    val linkList: List<LinkInfo> = emptyList()

) {

    companion object {
        fun of(linkInfoList: List<LinkInfo>): LinkListResult {
            return LinkListResult(
                linkList = linkInfoList
            )
        }
    }

}
